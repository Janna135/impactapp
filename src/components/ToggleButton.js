import React, { Component } from 'react'
import styled from 'react-emotion'
import { injectGlobal } from 'emotion'

injectGlobal(`
  * {
    box-sizing: border-box;
  }
`)

const StyledButton = styled('div')`
  background-color: ${props => (props.checked ? '#7AE582' : '#00A5CF')};
  border-radius: 5px;
  color: ${props => (props.checked ? '#004E64' : '#9fffcb')};
  display: flex;
  justify-content: center;
  align-items: center;
  margin-bottom: 12px;
  height: 45px;
  flex-shrink: 0;
`

export default class ToggleButton extends Component {
  render() {
    const { text, checked, onClick } = this.props

    return (
      <StyledButton checked={checked} onClick={onClick}>
        {text}
      </StyledButton>
    )
  }
}
