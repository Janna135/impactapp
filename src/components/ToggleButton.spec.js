import ToggleButton from './ToggleButton'
import { shallow } from 'enzyme'
import React from 'react'
import toJson from 'enzyme-to-json'

describe('ToggleButton', () => {
  let wrapper

  let mockFunction = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <ToggleButton text={'foo'} checked={false} onClick={mockFunction} />
    )
  })

  it('reacts on a click', () => {
    expect(mockFunction.mock.calls.length).toBe(0)
    wrapper.simulate('click')
    expect(mockFunction.mock.calls.length).toBe(1)
  })

  it('renders a ToggleButton', () => {
    expect(wrapper.html()).toContain('foo')
  })

  it('changes the color', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ checked: true })
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('has a matching snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
