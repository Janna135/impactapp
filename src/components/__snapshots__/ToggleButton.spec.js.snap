// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleButton changes the color 1`] = `
<Styled(div)
  checked={false}
  onClick={[Function]}
>
  foo
</Styled(div)>
`;

exports[`ToggleButton changes the color 2`] = `
<Styled(div)
  checked={true}
  onClick={[Function]}
>
  foo
</Styled(div)>
`;

exports[`ToggleButton has a matching snapshot 1`] = `
<Styled(div)
  checked={false}
  onClick={[Function]}
>
  foo
</Styled(div)>
`;
